pages/signup.tsx

```diff
@@ -1,3 +1,163 @@
+                                }
+                            />
+                        )}
+                        {min`sm` && (
+                            <Button {...buttonProps} aria-label={name}>
+                                <HStack justify="start" spacing={8}>
+                                    <Image src={source} style={{ height: "35px", width: "auto" }} />
+                                    <Text>Sign Up with {name}</Text>
+                                </HStack>
+                            </Button>
+                        )}
+                    </li>
+                ))}
+            </Grid>
+        </VStack>
+    );
+};
+
+type FormData = {
+    email: string;
+    username: string;
+    password: string;
+    term: boolean;
+};
+
+const ManualFormSignUp = () => {
+    const { register, handleSubmit, errors } = useForm<FormData>();
+    const onSubmit = (userData: any) => {
+        console.log(userData);
+    };
+
+    return (
+        <form tw="grid gap-4" onSubmit={handleSubmit(onSubmit)}>
+            <InputField
+                name="email"
+                label="Email address"
+                helperText="foobar@abc.com"
+                ref={register}
+                required
+            />
+            <InputField
+                name="username"
+                label="Username"
+                helperText="johndoe123"
+                ref={register}
+                required
+            />
+            <InputField
+                name="password"
+                label="Password"
+                type="password"
+                ref={register({ minLength: 8 })}
+                required
+            />
+            {errors.password && (
+                <span tw="flex justify-start text-xs text-gray-500">
+                    Password must be at least 8 chracters long
+                </span>
+            )}
+            <label tw="inline-grid gap-2 grid-flow-col justify-start items-center text-sm">
+                <input type="checkbox" name="term" tw="form-checkbox" ref={register} required />I
+                agree to the{" "}
+                <a href="" onClick={() => alert("Add link!!")}>
+                    privacy policy
+                </a>
+            </label>
+            <Button
+                w="full"
+                fontSize="sm"
+                size="lg"
+                type="submit"
+                colorScheme="purple"
+                // variant="outline"
+                leftIcon={<UserPlus size={24} />}>
+                Sign Up
+            </Button>
+            <Text fontSize="sm" justifySelf="right">
+                Do you have an account?{" "}
+                <Link href="/login">
+                    <a>login</a>
+                </Link>
+            </Text>
+        </form>
+    );
+};
+
 export default function Signup() {
-    return <div></div>;
+    return (
+        <Box
```

Comment:
Box is the most abstract component in Chakra UI, so I put it on the highest level.

Reply:
YES